### ISSUE TITLE
Rule Change: (fill in)

### ISSUE THREAD

**Issue by jpc-ae** [eslint/eslint#19062]
### What rule do you want to change?

no-constant-binary-expression

### What change do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
if ( true
  && condition1
  && condition2
  && condition3
  && ( false
    || condition4
    || condition5
    || condition6
    || condition7
    || condition8
    || condition9
  )
  && ( condition10 || condition11 || condition12 )
) {
  ...
}
```

### What does the rule currently do for this code?

Unexpected constant truthiness on the left-hand side of a `&&` expression.
Unexpected constant truthiness on the left-hand side of a `||` expression.

### What will the rule do after it's changed?

Explicit true before && and false before || should not trigger warn/error unless a new option (allowTrueAndFalseOr) is set to false (I argue that the default should be to allow this construct as it's very useful for diffing and doesn't cause any of the issues pointed out in the blog entry for this rule.)

### Participation

- [x] I am willing to submit a pull request to implement this change.

**Comment by nzakas** [comment]
It sounds like you're asking for a stylistic option to be added to the rule, but this rule isn't stylistic, so I don't think that makes sense.

I'd suggest using a disable comment for the cases where you want to disable this rule.

**Comment by jpc-ae** [comment]
I understand my use case is perhaps more stylistic in nature, but what I'm arguing is that this falls under a similar category to the while(true) option (https://github.com/eslint/eslint/issues/5477).

While something that evaluates to a constant condition is usually a bug, there are legitimate reasons to use the explicit `true &&` and `false ||` paradigms. I just don't want to have to turn this rule off for large conditionals where it's arguably more useful (what if one of the other conditions is constant?)
